node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -520,6 +520,10 @@ func (atp *apiTransactionProcessor) computeTimestampForRound(round uint64) int64
 	secondsSinceGenesis := round * atp.roundDuration
 	timestamp := atp.genesisTime.Add(time.Duration(secondsSinceGenesis) * time.Millisecond)
 
+	if atp.enableEpochsHandler.IsFlagEnabled(common.SupernovaFlag) {
```

Comment:
maybe this needs to be adapted with the merge from barnard?

Reply:
yes, in the merge from barnard, we have the handling for the new timestamp field for api.Block

Reply:
in the next PRs, we'll handle the activation flag check at transition